src/Service/ZaakService.php

```diff
@@ -391,148 +165,121 @@ private function checkZaakType(array $case)
      *
-     * @param array        $zaakArray      ZGW Zaak.
-     * @param array        $zaakTypeArray  ZGW ZaakType.
-     * @param ObjectEntity $zaakTypeObject ZGW ZaakType object.
-     * @param array        $case           xxllnc case object.
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\SyntaxError
      *
-     * @return array
+     * @return ObjectEntity The resulting zaak object.
      */
-    private function createSubObjects(array $zaakArray, array $zaakTypeArray, ObjectEntity $zaakTypeObject, array $case): array
+    public function syncCase(array $case, bool $flush = true): ObjectEntity
     {
-        if (isset($zaakTypeArray['statustypen']) === true && isset($case['instance']['milestone']) === true) {
-            $zaakArray = $this->mapStatus($zaakArray, $zaakTypeArray, $case['instance']['milestone']);
-        }
-
-        if (isset($zaakTypeArray['roltypen']) === true && isset($case['instance']['route']['instance']['role']) === true) {
-            $zaakArray = $this->mapRollen($zaakArray, $zaakTypeArray, $case['instance']['route']['instance']['role']);
-        }
-
-        if (isset($zaakTypeArray['eigenschappen']) === true && isset($case['instance']['attributes']) === true) {
-            $zaakArray = $this->mapEigenschappen($zaakArray, $zaakTypeArray, $zaakTypeObject, $case['instance']['attributes']);
-        }
-
-        return $zaakArray;
+        $zaakSchema  = $this->resourceService->getSchema(
+            'https://vng.opencatalogi.nl/schemas/zrc.zaak.schema.json',
+            'common-gateway/xxllnc-zgw-bundle'
+        );
+        $xxllncAPI   = $this->resourceService->getSource(
+            'https://development.zaaksysteem.nl/source/xxllnc.zaaksysteem.source.json',
+            'common-gateway/xxllnc-zgw-bundle'
+        );
+        $caseMapping = $this->resourceService->getMapping(
+            'https://development.zaaksysteem.nl/mapping/xxllnc.XxllncCaseToZGWZaak.mapping.json',
+            'common-gateway/xxllnc-zgw-bundle'
+        );
 
-    }//end createSubObjects()
-
-
-    /**
-     * Creates or updates a case to zaak.
-     *
-     * @param array $case Case from the Xxllnc API.
-     *
-     * @return void|null
-     */
-    public function caseToZaak(array $case, bool $flush = true)
-    {
-        $this->checkId($case);
         $zaakTypeObject = $this->checkZaakType($case);
         if ($zaakTypeObject instanceof ObjectEntity === false) {
-            isset($this->style) === true && $this->style->error("ZaakType for case {$case['reference']} could not be found or synced");
+            isset($this->style) === true && $this->style->error("ZaakType for case {$case['reference']} could not be found or synced, aborting.");
 
             return null;
         }
 
-        $zaakTypeArray = $zaakTypeObject->toArray();
+        isset($this->style) === true && $this->style->info("Mapping case to zaak..");
```

Comment:
Try to prevent using symfonyStyle, only use this if we can reach this code through a symfony command.
Always, add least use $This->logger for logging and maybe also include SymfonyStyle logging when needed

Reply:
there are commands for these functions